### ISSUE TITLE
Hairpins include one extra note than they should (in braille display)

### ISSUE THREAD

**Issue by HattieDouglass** [musescore/MuseScore#25203]
### Issue type

Engraving bug (incorrect score rendering)

### Description with steps to reproduce

Hairpins like crescendo and decrescendo include the note after the hairpin has ended

1. Create a score with, say, four B crotchets then a C semibreve
2. Select the four B crotchets and add a hairpin decrescendo
3. Select the C semibreve.
4. Press alt-f11 to turn on braille panel

<img width="173" alt="image" src="https://github.com/user-attachments/assets/603ba2fa-72fa-4dda-a2af-624ecd9159fe">

Actual behaviour: The braille panel shows the decrescendo end (⠜⠲) after the C semibreve (⠨⠽)
⠐⠺⠺⠺⠺   ⠨⠽⠜⠲
Expected behaviour: The braille panel should show the decrescendo end after the last of the four B crotchets (⠐⠺⠺⠺⠺)
⠐⠺⠺⠺⠺⠜⠲   ⠨⠽

### Supporting files, videos and screenshots

<img width="172" alt="image" src="https://github.com/user-attachments/assets/c79b2273-c84d-4ee3-b393-516f3af7075e">
<img width="170" alt="image" src="https://github.com/user-attachments/assets/e4072c95-2d1d-4cce-9983-da6f9cee743a">

### What is the latest version of MuseScore Studio where this issue is present?

OS: Windows 11 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.2-242570931, revision: github-musescore-musescore-3130f97

### Regression

I was unable to check

### Operating system

Windows 11

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by fcnjd** [comment]
See my PR #32680 that would fix this issue.
